src/features/unified-share-modal/HeaderTitleSection.js

```diff
@@ -0,0 +1,68 @@
+// @flow
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import Classification from '../classification';
+import messages from './messages';
+import type { item as ItemType } from './flowTypes';
+
+type Props = {
+    classification: ClassificationInfo,
+    isEmailLinkSectionExpanded: boolean,
+    item: ItemType,
+    showCollaboratorList: boolean,
+};
+
+const renderModalTitle = (isEmailLinkSectionExpanded, showCollaboratorList, item) => {
+    const { name } = item;
+    let title;
+
+    if (isEmailLinkSectionExpanded) {
+        title = (
+            <FormattedMessage
+                {...messages.emailModalTitle}
+                values={{
+                    itemName: name,
+                }}
+            />
+        );
+    } else if (showCollaboratorList) {
+        title = (
+            <FormattedMessage
+                {...messages.collaboratorListTitle}
+                values={{
+                    itemName: name,
+                }}
+            />
+        );
+    } else {
+        title = (
+            <FormattedMessage
+                {...messages.modalTitle}
+                values={{
+                    itemName: name,
+                }}
+            />
+        );
+    }
+
+    return title;
+};
+
+const HeaderTitleSection = ({ classification, isEmailLinkSectionExpanded, showCollaboratorList, item }: Props) => {
+    const title = renderModalTitle(isEmailLinkSectionExpanded, showCollaboratorList, item);
+    const { advisoryMessage, name } = classification;
+
+    return (
+        <span className="modal-title-label">
+            {title}
+            {advisoryMessage && (
+                <span className="EditClassificationButton">
```

Comment:
Is this advisoryMessage check and wrapping span needed?
Can you just have this below the title

`<Classification advisoryMessage={advisoryMessage} messageStyle="tooltip" name={name} />`

Reply:
I put this advisoryMessage check here is to avoid rendering the box if there is no classification assigned, and there is a margin-left style set on "EditClassificationButton" span.

Reply:
advisory message is not the classification though. Its the classification `name` that holds that value. The component internally takes care of what to show or not show, so check is not really that necessary. If you remove it, it should still work or at least have the name check. Or you can use the ClassifiedBadge component instead, that just shows the badge.

If you want styling, then use a class specific to this component. We are also not editing classification in this case.

Reply:
Thanks a lot! branch updated.